src/GrpcProtos/capture.proto

```diff
@@ -139,6 +139,148 @@ message ApiEvent {
   fixed64 r5 = 9;
 }
 
+message ApiScopeStart {
+  // NextID: 16
+
+  int32 pid = 1;
+  int32 tid = 2;
+  uint64 timestamp_ns = 3;
+
+  // We encode the first 64 characters of the scope's name/label using fixed64
+  // values, in order to avoid an additional allocation (for the internal
+  // vector) for short strings. Those allocations has shown a huge performance
+  // impact.
+  // If the name exceeds the 64 characters, the `encoded_name_additional` will
+  // be used to encode the additional characters.
+  fixed64 encoded_name_1 = 4;
+  fixed64 encoded_name_2 = 5;
+  fixed64 encoded_name_3 = 6;
+  fixed64 encoded_name_4 = 7;
+  fixed64 encoded_name_5 = 8;
+  fixed64 encoded_name_6 = 9;
+  fixed64 encoded_name_7 = 10;
+  fixed64 encoded_name_8 = 11;
+  bytes encoded_name_additional = 12;
+
+  Color color = 13;
+  uint64 group_id = 14;
+  uint64 address_in_function = 15;
+}
+
+message ApiScopeStop {
+  int32 pid = 1;
+  int32 tid = 2;
+  uint64 timestamp_ns = 3;
+}
+
+message ApiScopeStartAsync {
+  // NextID: 16
+
+  int32 pid = 1;
+  int32 tid = 2;
+  uint64 timestamp_ns = 3;
+
+  // We encode the first 64 characters of the scope's name/label using fixed64
+  // values, in order to avoid an additional allocation (for the internal
+  // vector) for short strings. Those allocations has shown a huge performance
+  // impact.
+  // If the name exceeds the 64 characters, the `encoded_name_additional` will
+  // be used to encode the additional characters.
+  fixed64 encoded_name_1 = 4;
+  fixed64 encoded_name_2 = 5;
+  fixed64 encoded_name_3 = 6;
+  fixed64 encoded_name_4 = 7;
+  fixed64 encoded_name_5 = 8;
+  fixed64 encoded_name_6 = 9;
+  fixed64 encoded_name_7 = 10;
+  fixed64 encoded_name_8 = 11;
+  bytes encoded_name_additional = 12;
+
+  Color color = 13;
```

Comment:
@dpallotti Should we then also get rid of the `Color` here in order to avoid allocations?

Reply:
Oooh yes please, I didn't think about it. Encoding as one uint32 is common and will suffice.